backend/src/main/java/com/woowacourse/gongseek/vote/application/VoteService.java

```diff
@@ -0,0 +1,144 @@
+import com.woowacourse.gongseek.vote.presentation.dto.VoteCreateRequest;
+import com.woowacourse.gongseek.vote.presentation.dto.VoteCreateResponse;
+import com.woowacourse.gongseek.vote.presentation.dto.VoteItemResponse;
+import com.woowacourse.gongseek.vote.presentation.dto.VoteResponse;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Transactional
+@Service
+public class VoteService {
+
+    private final VoteRepository voteRepository;
+    private final VoteItemRepository voteItemRepository;
+    private final ArticleRepository articleRepository;
+    private final MemberRepository memberRepository;
+    private final VoteHistoryRepository voteHistoryRepository;
+
+    public VoteCreateResponse create(AppMember appMember, Long articleId, VoteCreateRequest voteCreateRequest) {
+        Member member = getMember(appMember);
+
+        Article article = getArticle(articleId);
+        validateAuthor(member, article);
+        validateCategory(article);
+        Vote vote = voteRepository.save(new Vote(article, voteCreateRequest.getExpiryDate()));
+
+        Set<VoteItem> voteItems = VoteItems.of(voteCreateRequest.getItems(), vote).getVoteItems();
+        voteItemRepository.saveAll(voteItems);
+        return new VoteCreateResponse(vote);
+    }
+
+    private Member getMember(AppMember appMember) {
+        return memberRepository.findById(appMember.getPayload())
+                .orElseThrow(MemberNotFoundException::new);
+    }
+
+    private Article getArticle(Long articleId) {
+        return articleRepository.findById(articleId)
+                .orElseThrow(ArticleNotFoundException::new);
+    }
+
+    private void validateAuthor(Member member, Article article) {
+        if (!article.isAuthor(member)) {
+            throw new NoAuthorizationException();
+        }
+    }
+
+    private void validateCategory(Article article) {
+        if (article.cannotCreateVote()) {
+            throw new CannotCreateVoteException();
+        }
+    }
+
+    @Transactional(readOnly = true)
+    public VoteResponse getOne(Long articleId, AppMember appMember) {
+        if (!articleRepository.existsById(articleId)) {
+            throw new ArticleNotFoundException();
+        }
+        Vote foundVote = getVoteByArticleId(articleId);
+        List<VoteItem> voteItems = voteItemRepository.findAllByVoteArticleId(articleId);
+
+        VoteHistory voteHistory = voteHistoryRepository.findByVoteIdAndMemberId(foundVote.getId(),
+                appMember.getPayload()).orElse(null);
+
+        return new VoteResponse(foundVote, convertVoteItemResponse(voteItems), getVotedItemIdOrNull(voteHistory), foundVote.isExpired());
```

Comment:
foundVote 대신 파라미터로 받는 articleId를 넘겨주면 될 것 같은데 이렇게 구현하신 이유가 있나요?
VoteResponse에 필드가 articleId인데, vote를 넘기는게 이해 하는데 절차를 더 거치는 느낌이라 가독성을 위해서 수정하는 것은 어떤가요?
```suggestion
        return new VoteResponse(articleId, convertVoteItemResponse(voteItems), getVotedItemIdOrNull(voteHistory), foundVote.isExpired());
```

Reply:
articleId 로 생성된 투표가 있는지 확인해야 하기 때문에 foundVote 를 찾는 과정이 필요해요.
매개변수로 vote를 넘기는 부분은 불필요하다고 느껴져서 아래와 같이 vote에서 articleId 를 조회해서 넘기는 방법으로 변경했어요.
```java
return new VoteResponse(foundVote.getArticle().getId(), convertVoteItemResponse(voteItems), getVotedItemIdOrNull(voteHistory), foundVote.isExpired());
```